File: pingyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        List<TbBrand> tbBrands = brandService.findAll();
        return tbBrands;
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page, int rows) {
        PageResult page1 = brandService.findPage(page, rows);
        return page1;

    }

    //增加
    @RequestMapping("/findAdd")
    public Result findAdd(@RequestBody TbBrand tbBrand) {

        try {
            brandService.findAdd(tbBrand);
            return new Result(true, "添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "添加失败");
        }

    }

    //修改
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand) {
        try {
            brandService.update(tbBrand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    @RequestMapping("/delects")
    public Result delects(long[] id) {
        try {
            brandService.delects(id);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand, int page, int rows) {

        PageResult page1 = brandService.findPage(tbBrand, page, rows);
        return page1;

    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {

        return brandService.selectOptionList();
    }

}

File: pingyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandcontroller.js
app.controller("brandcontroller",function ($scope,$controller,brandService) {


    $controller('basecontroller',{$scope:$scope});//继承

    //读取列表数据绑定到表单中

    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    }
    //分页
    $scope.findPage=function (page,rows) {
        brandService.findPage(page,rows).success(

            function (response) {
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;
            }
        )

    }
    //查询实体
    $scope.findOne=function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
                $scope.reloadList();
            }
        )
    }
    //保存
    $scope.findAdd=function () {
        var serviceObject;
        if($scope.entity.id!=null){
            serviceObject=brandService.update($scope.entity)
        }else {
            serviceObject=brandService.findAdd($scope.entity)
        }
        serviceObject.success(
            function (response) {
                if (response.success){
                    //为ture 就重新加载查询页面
                    $scope.reloadList()
                }else {
                    alert(response.message);
                }
            }
        )
    }

    //批量删除
    $scope.dele=function () {

        brandService.dele($scope.selectIds).success(
            function (response) {
                if (response.success){
                    //为ture 就重新加载查询页面
                    $scope.reloadList()
                }else {
                    alert(response.message);
                }
            }
        )
    }
    $scope.y={}  //定义搜索对象
    //查询
    $scope.search=function (page,rows) {
        brandService.search(page,rows,$scope.y).success(

            function (response) {
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;
            }
        )

    }
})